packages/forms/src/UIForm/UIForm.component.js

```diff
@@ -40,23 +40,38 @@ export default class UIForm extends React.Component {
 	 * - onChange: for each field change
 	 * - onTrigger: when trigger is provided and its value is "after"
 	 * @param event The event that triggered the callback
-	 * @param payload { schema, value } The field schema and its new value
+	 * @param schema The payload field schema
+	 * @param value The payload new value
+	 * @param deepValidation Validate the subItems
+	 * @param widgetChangeErrors Change errors hook, allows any widget to manipulate the errors map
 	 */
-	onChange(event, { schema, value }) {
-		const {
-			formName,
-			onChange,
-			properties,
-			customValidation,
-		} = this.props;
-		const error = validateValue(schema, value, properties, customValidation);
-		const payload = { formName, schema, value, error, properties };
-		onChange(event, payload);
-
-		const { triggers } = schema;
-		if (triggers && triggers.includes(TRIGGER_AFTER)) {
+	onChange(event, { schema, value }, { deepValidation = false, widgetChangeErrors } = {}) {
+		const error = validateSingle(
+			schema,
+			value,
+			this.props.properties,
+			this.props.customValidation,
+			deepValidation
+		)[schema.key];
+
+		const payload = {
+			formName: this.props.formName,
+			properties: this.props.properties,
+			schema,
+			value,
+			error,
+		};
+		this.props.onChange(event, payload);
+
+		if (schema.triggers && schema.triggers.includes(TRIGGER_AFTER)) {
 			this.onTrigger(event, { type: TRIGGER_AFTER, ...payload });
 		}
+
+		if (widgetChangeErrors) {
+			const errors = Object.assign(widgetChangeErrors(this.props.errors));
```

Comment:
Is it here to perform a copy of the object ? If it is, this should be 

`const errors = Object.assign({}, widgetChangeErrors(this.props.errors));`